### PR TITLE
ESLint update

### PR DESCRIPTION
**Pull request by chrmod**

### CODE REVIEW

**Comment by chrmod** [comment]
Changes work fine in CI, but VSCode still don't recognise the new setup and is unable to provide eslint based suggestions. Need to figure it out before merging.

File: eslint.config.js
Comment (JoshuaKGoldberg) [inline eslint.config.js:1]: > but VSCode still don't recognise the new setup and is unable to provide eslint based suggestions

See https://github.com/microsoft/vscode-eslint/issues/1644 for flat config support. I'm guessing you probably want to set something like `eslint.experimental.useFlatConfig` / `eslint.useFlatConfig` in a `.vscode/settings.json` and/or your editor user settings.

Comment (chrmod) [reply eslint.config.js:1]: This works, fantastic! Many thanks!

File: package.json [hunk 1]
```diff
@@ -69,21 +69,21 @@
...
-    "eslint": "^8.49.0",
+    "eslint": "^9.3.0",
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-prettier": "^5.0.0",
     "lerna": "^8.0.0",
     "patch-package": "^8.0.0",
     "prettier": "^3.0.3",
-    "typescript": "^5.4.5"
+    "typescript": "^5.4.5",
+    "typescript-eslint": "8.0.0-alpha.16"
```
Comment (remusao) [inline package.json:86]: Not stable yet?

File: package.json [hunk 2]
```diff
@@ -69,21 +69,21 @@
     "format-fix": "prettier './packages/**/*.ts' --write"
   },
   "devDependencies": {
+    "@eslint/js": "^9.3.0",
```
Comment (remusao) [inline package.json:72]: What does this do?

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      '@typescript-eslint/ban-ts-comment': 'off',
+      '@typescript-eslint/no-redundant-type-constituents': 'off',
+      '@typescript-eslint/no-duplicate-type-constituents': 'off',
+      '@typescript-eslint/no-unsafe-argument': 'off',
+      '@typescript-eslint/no-floating-promises': 'off',
+      '@typescript-eslint/no-unsafe-return': 'off',
+      '@typescript-eslint/no-misused-promises': 'off',
+      '@typescript-eslint/require-await': 'off',
+      '@typescript-eslint/no-unused-vars': 'off',
+      '@typescript-eslint/no-empty-object-type': 'off',
```
Comment (remusao) [inline eslint.config.js:35]: Would be nice to start fixing those

File: tsconfig.eslint.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "compilerOptions": {
+    "types": ["@types/node"],
+    "noEmit": true,
+    "allowJs": true
```
Comment (remusao) [inline tsconfig.eslint.json:5]: Do we have javascript files in the source tree?

**Review by remusao — APPROVED** [review]

File: tsconfig.eslint.json (hunk 4 above)
Comment (chrmod) [reply tsconfig.eslint.json:5]: this is just for the `eslint.config.js`

File: eslint.config.js (hunk 3 above)
Comment (chrmod) [reply eslint.config.js:35]: yes, had to add those as they came with the latest update of `typescript-eslint`. will address this in separate PRs

File: package.json (hunk 2 above)
Comment (chrmod) [reply package.json:72]: this is the part of new setup - https://eslint.org/docs/latest/use/getting-started#manual-set-up

Perhaps we can remove other package from the package.json. Will check

File: package.json (hunk 1 above)
Comment (chrmod) [reply package.json:86]: on the way, but we can confirm it works https://github.com/typescript-eslint/typescript-eslint/issues/8211

**Pull request merged**